src/main/java/org/tugraz/sysds/runtime/instructions/fed/AggregateBinaryFEDInstruction.java

```diff
@@ -17,6 +17,7 @@
 package org.tugraz.sysds.runtime.instructions.fed;
 
 import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.MutablePair;
```

Comment:
could it be done with the Immutable Pair?

Reply:
Maybe by rewriting my multi-threading, but I don't expect any large gains. Could you elaborate?